### ISSUE TITLE
`import/named` no error on bad import in JS file from TS file

### ISSUE THREAD

**Issue by aaronadamsCA** [import-js/eslint-plugin-import#2235]
We are gradually converting a large Gatsby codebase from JS to TS.

We accidentally tried to import a non-existent named export from a TS file into a JS file:

```js
import { price } from "./price-variants";
```

There was no such named export in `price-variants.ts`, but `import/named` didn't report it.

I'm sure it might complain if there were a transpiled `price-variants.js`, but Gatsby projects use `"noEmit": true`. The only chance we really have to catch a bad import before runtime, then, is with this plugin.

My guess is this might be a known limitation, but it makes it difficult to gradually convert a large codebase from JS to TS. Ideally each parser would parse its own files, and import names could be checked across parsers.

<details><summary><code>.eslintrc.cjs</code></summary>

(Our actual config is much larger, but I've confirmed the cut-down version below still exhibits the same behaviour.)

```js
module.exports = {
  root: true,
  extends: [
    "eslint:recommended",
    "plugin:react/recommended",
    "plugin:import/recommended",
  ],
  parser: "@babel/eslint-parser",
  parserOptions: {
    sourceType: "module",
  },
  env: {
    browser: true,
    es2020: true,
    node: true,
  },
  rules: {
    "react/jsx-uses-react": "off",
    "react/react-in-jsx-scope": "off",
    "react/prop-types": "off",
  },
  overrides: [
    {
      // Lint all files with any of these extensions
      files: ["*.cjs", "*.js", "*.jsx", "*.ts", "*.tsx"],
    },
    {
      files: ["*.ts", "*.tsx"],
      extends: [
        "plugin:@typescript-eslint/recommended",
        "plugin:import/typescript",
      ],
      parser: "@typescript-eslint/parser",
      parserOptions: {
        sourceType: "module",
      },
    },
  ],
  settings: {
    react: {
      version: "detect",
    },
    // Doesn't work; causes ESLint parse errors in JS files that import TS files
    // "import/extensions": [".js", ".jsx", ".ts", ".tsx"],
    "import/resolver": {
      node: {
        extensions: [".js", ".jsx", ".ts", ".tsx"],
      },
    },
  },
};
```

</details>

Happy to try different settings. If we can't find anything that works, maybe I can put together a minimal repro.

**Comment by ljharb** [comment]
What you're missing is the `import/resolver` needs to also use the typescript resolver.

**Comment by aaronadamsCA** [comment]
I've tried that, unfortunately it triggers a "Resolve error: typescript with invalid interface loaded as resolver" error. I'm not sure where the issue is; I've installed `eslint-import-resolver-typescript`, so I have no idea why the plugin sees it as invalid.

That said, I did find a solution, I was evidently missing this in `settings`:

```js
    "import/parsers": {
      "@typescript-eslint/parser": [".ts", ".tsx"],
    },
```

Our project evidently has no problem using the Node.js resolver, but we will still probably need the TypeScript resolver in the future. If you have any idea why that might be failing, please let me know. I'll close this issue as resolved.

**Comment by aaronadamsCA** [comment]
...When in doubt, nuke `node_modules/` and try again! It works, thank you. I needed both to be more specific with `import/parsers` as well as what you said about `import/resolver`.

I'll post the final working config in a moment.

**Comment by aaronadamsCA** [comment]
Fully functioning ESLint config for a mixed JS/TS Gatsby codebase:

<details><summary><code>.eslintrc.cjs</code></summary>

```js
module.exports = {
  root: true,
  parser: "@babel/eslint-parser",
  extends: [
    "eslint:recommended",
    "plugin:react/recommended",
    "plugin:import/recommended",
  ],
  parserOptions: {
    sourceType: "module",
  },
  env: {
    browser: true,
    es2020: true,
    node: true,
  },
  rules: {
    "react/jsx-uses-react": "off",
    "react/react-in-jsx-scope": "off",
    "react/prop-types": "off",
  },
  overrides: [
    {
      // Make sure ESLint can see every possible Node/TypeScript file
      files: ["*.cjs", "*.js", "*.jsx", "*.mjs", "*.ts", "*.tsx"],
    },
    {
      files: ["*.ts", "*.tsx"],
      extends: [
        "plugin:@typescript-eslint/recommended",
        "plugin:import/typescript",
      ],
    },
  ],
  settings: {
    react: {
      version: "detect",
    },
    "import/parsers": {
      "@typescript-eslint/parser": [".ts", ".tsx"],
    },
    "import/resolver": {
      typescript: {
        project: "packages/*/tsconfig.json",
      },
      node: {},
      alias: [["@reach/router", "@gatsbyjs/reach-router"]],
    },
  },
};
```

</details>
